### DISCUSSION TITLE
Pandoc Full Support For JATSXML Specification?

### DISCUSSION THREAD

**Discussion by aerithnetzer** [jgm/pandoc#9818] [Q&A] [answered]
Hello,

This may be a simple question, we are currently building a product that will aid in the discovery of data and code affiliated with academic articles. I am unsure if Pandoc currently supports the full JATSXML [code specification](https://jats.nlm.nih.gov/publishing/tag-library/1.3/element/code.html). It seems in the [documentation](https://pandoc.org/jats.html) that some elements are not writable. Does there happen to be any work/documentation on this? I am happy to help with a PR if needed.

Thanks!

**Comment by jgm** [comment]
First it would help to know whether you are talking about *reading* JATS or writing it.  I assume you're talking about reading JATS?  The pandoc AST is strictly less expressive than JATSXML, so inevitably some distinctions will be lost when reading JATS with pandoc.

**Reply by aerithnetzer** [reply]
Thanks for the reply. I am actually talking mostly about writing JATS. We would ideally like to take something like a markdown file with a header including something like 

```yaml 
code:
    code-type:
    code-version:
```
And write the required `code` tag to the XML with the relevant attributes. If Pandoc AST does not include these distinctions I would assume something like a filter would be more appropriate?

Thanks.

**Reply by jgm** [reply]
When you say `code` tag you mean https://jats.nlm.nih.gov/archiving/tag-library/1.2/element/code.html ?
I'm confused, because this is normally an element of the document's body, and not something it would make sense to specify in metadata.

**Reply by aerithnetzer** [reply]
I may be misunderstanding the JATS specification, so forgive me if I am. Yes, the `code` element link you refer to is the correct tag I am referring to. But, according to the documentation says that the `code` element may be contained in the `supplementary-material` tag, which itself may be contained in the `article-meta` tag. To me, this means that the `code` element you refer to can in fact be part of the metadata. For our library's use case (attempting to create articles that are interlinked with the code and data used in the article), code that is meant to supplement the article, say, for reproducibility, it does make sense to use it as metadata. Given this, is there a way to represent this in markdown YAML?

Hopefully this makes more sense/context.

**Reply by jgm** [reply]
That helps.  Can you give a short snippet of the sort of XML you're hoping to generate?

**Reply by aerithnetzer** [reply]
Hello! Sorry for delayed reply, out of town. Hope you had a good weekend.

```xml
      
<!-- ...XML header info-->
<article-meta>      
      <supplementary-material>
        <!-- Other supplementary materials-->
        <code>
          <!-- Link to github url where repo can be used to aid in study reproducibility, etc. -->
          <ext-link ext-link-type="uri" xlink:href="github.com/aerithnetzer/somegithubrepo"></ext-link>
        </code>
    </supplementary-material>
</article-meta>
<!-- Rest of the document... -->
```

This is what I am hoping to generate.

**Reply by jgm** [reply, marked as answer]
As far as I can tell, `supplementary-material` is only allowed in `article-meta`, not `journal-meta`, so I will add it to the default template only for `-t jats_articleauthoring`.

Once it is added, you'll be able to get what you want with `pandoc -t jats_articleauthoring -s` using:
````
---
title: My Title
supplementary-material: |
  ```{=jats}               
  <code>
    <ext-link ext-link-type="uri" xlink:href="github.com/aerithnetzer/somegithubrepo"></ext-link>
  </code>
  ```
...
````

**Reply by aerithnetzer** [reply]
Awesome, thanks a ton
